bigquery_etl/query_scheduling/dags.py

```diff
@@ -0,0 +1,70 @@
+"""Represents a collection of configured Airflow DAGs."""
+
+from itertools import groupby
+import yaml
+
+from bigquery_etl.query_scheduling.dag import Dag
+
+
+class Dags:
+    """Representation of all configured DAGs."""
+
+    def __init__(self, dags):
+        """Instantiate DAGs."""
+        self.dags = dags
+
+    @classmethod
+    def from_dict(cls, d):
+        """
+        Parse DAG configurations from a dict and create new instances.
+
+        Expected dict format:
+        {
+            "dag_name1": {
+                "schedule_interval": string,
+                "default_args": dict
+            },
+            "dag_name2": {
+                "schedule_interval": string,
+                "default_args": dict
+            },
+            ...
+        }
+        """
+        dags = [Dag.from_dict({k: v}) for k, v in d.items()]
+        return cls(dags)
+
+    @classmethod
+    def from_file(cls, config_file):
+        """Instantiate DAGs based on the provided configuration file."""
+        with open(config_file, "r") as yaml_stream:
+            dags_config = yaml.safe_load(yaml_stream)
+            return Dags.from_dict(dags_config)
+
+    def dag_by_name(self, name):
+        """Return the DAG with the provided name."""
+        for dag in self.dags:
+            if dag.name == name:
+                return dag
+
+        return None
+
+    def with_tasks(self, tasks):
+        """Assign tasks to their corresponding DAGs."""
+        for dag_name, tasks in groupby(tasks, lambda t: t.dag_name):
+            dag = self.dag_by_name(dag_name)
+
+            if dag is None:
+                print(
```

Comment:
Should we raise an exception instead? Or perhaps set a flag so that we can return a failure state later?